### PR TITLE
Torch mnist doc

### PR DESCRIPTION
**Pull request by pedropgusmao**
Created the MNIST Pytorch Quickstart example including documentation with numpydoc style.

### CODE REVIEW

File: src/py/flwr_example/quickstart_pytorch/client.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    parser.add_argument(
+        "--server_address",
+        type=str,
+        default="[::]:8080",
+        help=f"gRPC server address (default: '[::]:8080')",
+    )
+    parser.add_argument(
+        "--cid",
+        type=int,
+        default=0,
```
Comment (danieljanes) [inline src/py/flwr_example/quickstart_pytorch/client.py:39]: I'd suggest to better remove the default here.

File: src/py/flwr_example/quickstart_pytorch/mnist.py [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+    Parameters
+    ----------
+    dataset: torch.utils.data.Dataset
+        Dataset to be partitioned into *number_of_clients* subsets.
+        
+    batch_size: int 
+        Size of mini-batches used by the returned DataLoader.
+        
+    client_id: int 
+        Unique integer used for selecting a specif partition.
```
Comment (danieljanes) [inline src/py/flwr_example/quickstart_pytorch/mnist.py:61]: Small typo: `specific`

File: src/py/flwr_example/quickstart_pytorch/mnist.py [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+        Mini-batch size for training set.
+        
+    test_batch_size: int 
+        Mini-batch size for test set.
+        
+    cid: int 
+        Client ID used to select a specific partition.
+        
+    nb_clients: int 
+        Total number of clients launched during traning. This value dictates the number of unique to be created.
```
Comment (danieljanes) [inline src/py/flwr_example/quickstart_pytorch/mnist.py:118]: Small typo: `training`

File: src/py/flwr_example/quickstart_pytorch/mnist.py [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+        dataset=test_dataset,
+        batch_size=test_batch_size,
+        client_id=cid,
+        number_of_clients=nb_clients,
+    )
+
+    return (train_loader, test_loader)
+
+
+class MNISTNet(nn.Module):
```
Comment (danieljanes) [inline src/py/flwr_example/quickstart_pytorch/mnist.py:155]: Can this be `class MNISTNet(nn.ModuleList)` to align with the type of the first parameter for `train(...)`?

File: src/py/flwr_example/quickstart_pytorch/mnist.py [hunk 5]
```diff
@@ -0,0 +1,434 @@
...
+) -> int:
+    """Train routine based on 'Basic MNIST Example'
+
+    Parameters
+    ----------
+    model: torch.nn.ModuleList
+        Neural network model used in this example.
+        
+    train_loader: torch.utils.data.DataLoader
+        DataLoader used in traning.
```
Comment (danieljanes) [inline src/py/flwr_example/quickstart_pytorch/mnist.py:212]: Typo: `training`

File: src/py/flwr_example/quickstart_pytorch/mnist.py [hunk 6]
```diff
@@ -0,0 +1,434 @@
...
+        Number of epochs to run in each round. 
+        
+    device: torch.device 
+         (Default value = torch.device("cpu"))
+         Device where the network will be trained within a client.
+
+    Returns
+    -------
+    num_examples_train: int
+        Number of total samples used during traning.
```
Comment (danieljanes) [inline src/py/flwr_example/quickstart_pytorch/mnist.py:224]: Typo: `training`

File: src/py/flwr_example/quickstart_pytorch/mnist.py [hunk 7]
```diff
@@ -0,0 +1,434 @@
...
+
+    """
+    model.train()
+    optimizer = optim.Adadelta(model.parameters(), lr=1.0)
+    scheduler = StepLR(optimizer, step_size=1, gamma=0.7)
+    print(f"Training {epochs} epoch(s) w/ {len(train_loader)} mini-batches each")
+    for epoch in range(epochs):  # loop over the dataset multiple time
```
Comment (danieljanes) [inline src/py/flwr_example/quickstart_pytorch/mnist.py:231]: Typo: `# .... times`

File: src/py/flwr_example/quickstart_pytorch/mnist.py [hunk 8]
```diff
@@ -0,0 +1,434 @@
...
+        print()
```
Comment (danieljanes) [inline src/py/flwr_example/quickstart_pytorch/mnist.py:232]: Is this empty print necessary?

File: src/py/flwr_example/quickstart_pytorch/mnist.py [hunk 9]
```diff
@@ -0,0 +1,434 @@
...
+        state_dict = OrderedDict(
+            {
+                k: torch.Tensor(v)
+                for k, v in zip(self.model.state_dict().keys(), weights)
+            }
+        )
+        self.model.load_state_dict(state_dict, strict=True)
+
+    def get_parameters(self) -> fl.common.ParametersRes:
+        """Encapsulates the weight into Flower Parameters """
```
Comment (danieljanes) [inline src/py/flwr_example/quickstart_pytorch/mnist.py:358]: Typo: `weights`

**Review by danieljanes — CHANGES_REQUESTED** [review]

File: src/py/flwr_example/quickstart_pytorch/mnist.py (hunk 8 above)
Comment (pedropgusmao) [reply src/py/flwr_example/quickstart_pytorch/mnist.py:232]: The print in line 250 updates the line. This print is not really necessary but it keeps the terminal pretty. Should I delete it?

File: src/py/flwr_example/quickstart_pytorch/server.py [hunk 10]
```diff
@@ -0,0 +1,4 @@
+import flwr as fl
```
[the remark this answers is not in the document]
Comment (pedropgusmao) [reply src/py/flwr_example/quickstart_pytorch/server.py:15]: Resolved in b1559aeb7d39137b1bdc8f9d5ce5a9dc3a8ca43d

File: src/py/flwr_example/quickstart_pytorch/mnist.py.patch [hunk 11]
```diff
@@ -0,0 +1,184 @@
+# Patch generated by Pyment v0.3.3
```
[the remark this answers is not in the document]
Comment (pedropgusmao) [reply src/py/flwr_example/quickstart_pytorch/mnist.py.patch]: Yes, the file was removed in b1559aeb7d39137b1bdc8f9d5ce5a9dc3a8ca43d

File: src/py/flwr_example/quickstart_pytorch/mnist.py (hunk 2 above)
Comment (pedropgusmao) [reply src/py/flwr_example/quickstart_pytorch/mnist.py]: Resolved in b1559aeb7d39137b1bdc8f9d5ce5a9dc3a8ca43d

File: src/py/flwr_example/quickstart_pytorch/client.py (hunk 1 above)
Comment (pedropgusmao) [reply src/py/flwr_example/quickstart_pytorch/client.py]: Resolved in b1559aeb7d39137b1bdc8f9d5ce5a9dc3a8ca43d

File: src/py/flwr_example/quickstart_pytorch/mnist.py (hunk 4 above)
Comment (pedropgusmao) [reply src/py/flwr_example/quickstart_pytorch/mnist.py:155]: ModuleList was not necessary so I changed the other ones in train and test in b1559aeb7d39137b1bdc8f9d5ce5a9dc3a8ca43d .

File: src/py/flwr_example/quickstart_pytorch/mnist.py (hunk 5 above)
Comment (pedropgusmao) [reply src/py/flwr_example/quickstart_pytorch/mnist.py]: Solved in b1559aeb7d39137b1bdc8f9d5ce5a9dc3a8ca43d .

File: src/py/flwr_example/quickstart_pytorch/mnist.py (hunk 6 above)
Comment (pedropgusmao) [reply src/py/flwr_example/quickstart_pytorch/mnist.py]: Solved in b1559aeb7d39137b1bdc8f9d5ce5a9dc3a8ca43d .

File: src/py/flwr_example/quickstart_pytorch/mnist.py (hunk 7 above)
Comment (pedropgusmao) [reply src/py/flwr_example/quickstart_pytorch/mnist.py]: Solved in b1559aeb7d39137b1bdc8f9d5ce5a9dc3a8ca43d

File: src/py/flwr_example/quickstart_pytorch/mnist.py (hunk 9 above)
Comment (pedropgusmao) [reply src/py/flwr_example/quickstart_pytorch/mnist.py]: Solved in b1559aeb7d39137b1bdc8f9d5ce5a9dc3a8ca43d .

File: src/py/flwr_example/quickstart_pytorch/mnist.py (hunk 3 above)
Comment (pedropgusmao) [reply src/py/flwr_example/quickstart_pytorch/mnist.py]: Solved in 600102cbbb9579ad7a8add0a14d5e24d9b0f5f7c .

File: src/py/flwr_example/quickstart_pytorch/mnist.py (hunk 8 above)
Comment (danieljanes) [reply src/py/flwr_example/quickstart_pytorch/mnist.py:232]: Ah I see, we can leave it like that

**Review by danieljanes — APPROVED** [review]

**Pull request merged**